pulsar-common/src/main/java/org/apache/pulsar/common/api/Commands.java

```diff
@@ -168,6 +205,52 @@ public static ByteBuf newConnected(int clientProtocolVersion) {
         return res;
     }
 
+    public static ByteBuf newAuthChallenge(String authMethod, AuthData brokerData, int clientProtocolVersion) {
+        CommandAuthChallenge.Builder challengeBuilder = CommandAuthChallenge.newBuilder();
+
+        // If the broker supports a newer version of the protocol, it will anyway advertise the max version that the
+        // client supports, to avoid confusing the client.
+        int currentProtocolVersion = getCurrentProtocolVersion();
+        int versionToAdvertise = Math.min(currentProtocolVersion, clientProtocolVersion);
+
+        challengeBuilder.setProtocolVersion(versionToAdvertise);
+        challengeBuilder.setServerVersion("Pulsar Server");
```

Comment:
We should have the server version available somewhere. There's no point in just sending "Pulsar Server"

Reply:
Seems this is the same as in CommandConnected, the serverVersion is not verified. would like to remove this, and if needed, we could add it later if it is really need. This suits the `optional` in proto for compact.